OpenRA.Game/Widgets/Widget.cs

```diff
@@ -284,8 +284,15 @@ public virtual Rectangle GetEventBounds()
 			// PERF: Avoid LINQ.
 			var bounds = EventBounds;
 			foreach (var child in Children)
+			{
 				if (child.IsVisible())
-					bounds = Rectangle.Union(bounds, child.GetEventBounds());
+				{
+					var childBounds = child.GetEventBounds();
+					if (childBounds != Rectangle.Empty)
+						bounds = Rectangle.Union(bounds, child.GetEventBounds());
```

Comment:
Any reason to use `child.GetEventBounds()` again instead of passing the `childBounds` variable?